### DISCUSSION TITLE
Make the ordering case insensitive

### DISCUSSION THREAD

**Discussion by Abedrane** [alirezanet/Gridify#230] [Q&A]
Hello everyone, is there a way to make the ordering case insensitive when ordering based on a string property ? Correct me if I am wrong but the mappers only apply to filtering, right ? 

I have a bunch of file names to filter and order and the output is like this : 

- Banana.pdf
- Yellow.pdf
- peach.pdf

Whereas I would want something like : 

- Banana.pdf
- peach.pdf
- Yellow.pdf

Is there any way to achieve such a scenario ? Thank you !

**Comment by alirezanet** [comment]
Hi @Abedrane,
Ordering uses the same mapping profile as filtering so it should work as expected, do you get any exceptions?

**Reply by alirezanet** [reply]
looks ok to me: 
``` csharp
string[] files = ["Yellow.pdf", "Banana.pdf", "peach.pdf", "test.txt"];
var ds = files.Select(f => new FileNames(f)).AsQueryable();


var result = ds
	.ApplyFiltering("fileName=*.pdf")
	.ApplyOrdering("fileName");


result.Dump(); 
// Banana.pdf
// peach.pdf
// Yellow.pdf


record FileNames(string fileName);
```

**Reply by Abedrane** [reply]
Hi, thank you very much for your reply !

I made some tests on my side, and indeed it works. For my particular use-case the culprit is my database : CosmosDB. After looking on the internet it does not support case insensitive ordering, so this might be the culprit. I will have more time tomorrow to check how exactly the query is made on CosmosDB side and see how to work with it. 

Apologies for the confusion due to the lack of context. 

Thank you again and have a nice day !
